frontends/common/resolveReferences/resolveReferences.cpp

```diff
@@ -180,6 +180,7 @@ ResolveReferences::ResolveReferences(ReferenceMap* refMap,
 void ResolveReferences::addToContext(const IR::INamespace* ns) {
     LOG1("Adding to context " << dbp(ns));
     BUG_CHECK(context != nullptr, "No resolution context; did not start at P4Program?");
+    checkShadowing(ns);
     context->push(ns);
```

Comment:
It seems like this will now check *every* case of shadowing.  Are there cases where shadowing should not be a warning?  Actions in a control with the same name as some non-action global thing?  Cases where a programmer might reasonably expect to reuse a name?